app/src/main/java/com/itsaky/androidide/services/GradleBuildService.java

```diff
@@ -390,12 +390,10 @@ public boolean isBuildInProgress() {
   }
 
   public void startToolingServer(@Nullable OnServerStartListener listener) {
-    if (toolingServerThread != null && toolingServerThread.isAlive()) {
-      throw new ToolingServerAlreadyStartedException();
+    if (toolingServerThread == null) {
+      toolingServerThread = new Thread(new ToolingServerRunner(listener));
+      toolingServerThread.start();
     }

```

Comment:
This condition should at least check if the thread is alive or not and start the tooling server only if it is not alive.

Reply:
> This condition should at least check if the thread is alive or not and start the tooling server only if it is not alive.

But what will happen if the first condition (if Tooling is active), in which case it would return the exception, but it would be the same as before, so leave it empty?